build: add LD_BUILD_EXPORT_ALL_SYMBOLS option to export all symbols (#420)
launchdarkly/cpp-sdks: CMakeLists.txt

    set(Boost_USE_STATIC_LIBS ON)
endif ()

> if (LD_BUILD_SHARED_LIBS AND (NOT LD_BUILD_EXPORT_SYMBOLS))

Reviewer: ```suggestion
if (LD_BUILD_SHARED_LIBS AND (NOT LD_BUILD_EXPORT_CPP_SYMBOLS))
```

> if (LD_BUILD_SHARED_LIBS AND (NOT LD_BUILD_EXPORT_SYMBOLS))
>     # When building a shared library we hide all symbols
>     # aside from this we have specifically exported for the C-API.
>     message(STATUS "LaunchDarkly: hide all symbols aside from C-API")
>     set(CMAKE_CXX_VISIBILITY_PRESET hidden)
>     set(CMAKE_VISIBILITY_INLINES_HIDDEN 1)
> else ()
>     message(STATUS "LaunchDarkly: expose all symbols")
> endif ()

Reviewer: ```suggestion
if (LD_BUILD_SHARED_LIBS)
    if (LD_BUILD_EXPORT_ALL_SYMBOLS)
       message(STATUS "LaunchDarkly: exposing all symbols in shared libraries")
    else ()
       message(STATUS "LaunchDarkly: hiding all symbols in shared libraries except for C API")
       set(CMAKE_CXX_VISIBILITY_PRESET hidden)
       set(CMAKE_VISIBILITY_INLINES_HIDDEN 1)
    endif()
endif()
```
